Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -933,6 +1024,114 @@ void UpdatePlanes(std::vector<Frame::Plane::Identifier>& updatedPlanes, std::vec
             }
         }
 
+        /**
+         Updates existing meshes in place, gets the list of updated/created mesh IDs and removed mesh IDs.
+         */
+        void UpdateMeshes(std::vector<Frame::Mesh::Identifier>& updatedMeshes, std::vector<Frame::Mesh::Identifier>& deletedMeshes) {
+            if (!meshDetectionEnabled)
+            {
+                return;
+            }
+            
+            [sessionDelegate LockMeshes];
+            @try {
+                if (@available(iOS 13.4, *)) {
+                    // Update all meshes that have been updated since the last frame
+                    auto updatedARKitMeshes = [sessionDelegate GetUpdatedMeshes];
+                    for (ARMeshAnchor* updatedMesh : *updatedARKitMeshes) {
+                        auto geometry = updatedMesh.geometry;
+                        auto faces = geometry.faces;
+                        auto vertices = geometry.vertices;
+                        auto normals = geometry.normals;
+                        // get vertices data for vertex buffer
+                        meshVertexBuffer.clear();
+                        meshVertexBuffer.resize(vertices.count);
+                        std::vector<Vector3f> vertexData{};
+                        vertexData.resize(vertices.count);
+                        memcpy(vertexData.data(), vertices.buffer.contents, vertices.stride * vertices.count);
+                        for(int i = 0; i < vertices.count; i++) {
+                            auto vertex = vertexData[i];
+                            auto vertexTransform = matrix_identity_float4x4;
+                            vertexTransform.columns[3] = simd_float4{vertex.X, vertex.Y, vertex.Z, 1};
+                            auto transform = matrix_multiply(updatedMesh.transform, vertexTransform);
+                            auto position = simd_float3{transform.columns[3].x ,transform.columns[3].y, transform.columns[3].z};
+                            meshVertexBuffer[i].X = position[0];
+                            meshVertexBuffer[i].Y = position[1];
+                            meshVertexBuffer[i].Z = position[2];
+                        }
+                        // get indices data for index buffer
+                        auto indexCount = faces.count * faces.indexCountPerPrimitive;
```

Comment:
>auto [](http://example.com/codeflow?start=24&length=4)

nit: consider with some of these autos you are creating declaring them const if they aren't ever reupdated. const specifiers should be better for performance